cli/command/deploy.go

```diff
@@ -1,15 +1,169 @@
 package command
 
-import "github.com/urfave/cli"
+import (
+	"fmt"
+	"github.com/quintilesims/layer0/common/models"
+	"github.com/urfave/cli"
+	"io/ioutil"
+	"strconv"
+)
 
 type DeployCommand struct {
 	*CommandBase
 }
 
 func NewDeployCommand(b *CommandBase) *DeployCommand {
-	return &DeployCommand{b}
+	return &DeployCommand{
+		CommandBase: b,
+	}
 }
 
 func (d *DeployCommand) Command() cli.Command {
-	return cli.Command{}
+	return cli.Command{
+		Name:  "deploy",
+		Usage: "manage layer0 deploys",
+		Subcommands: []cli.Command{
+			{
+				Name:      "create",
+				Usage:     "create a new deploy",
+				Action:    d.create,
+				ArgsUsage: "PATH NAME",
+			},
+			{
+				Name:      "delete",
+				Usage:     "delete a deploy",
+				ArgsUsage: "NAME",
+				Action:    d.delete,
+			},
+			{
+				Name:      "get",
+				Usage:     "describe a deploy",
+				Action:    d.read,
+				ArgsUsage: "NAME",
+			},
+			{
+				Name:      "list",
+				Usage:     "list all deploys (only the latest versions of each family will be shown)",
+				Action:    d.list,
+				ArgsUsage: " ",
+				Flags: []cli.Flag{
+					cli.BoolFlag{
+						Name:  "all",
+						Usage: "list all versions of all deploys",
+					},
+				},
+			},
+		},
+	}
+}
+
+func (d *DeployCommand) create(c *cli.Context) error {
+	args, err := extractArgs(c.Args(), "PATH", "NAME")
+	if err != nil {
+		return err
+	}
+
+	content, err := ioutil.ReadFile(args["PATH"])
+	if err != nil {
+		return err
+	}
+
+	req := models.CreateDeployRequest{
+		DeployName: args["NAME"],
+		DeployFile: content,
+	}
+
+	jobID, err := d.client.CreateDeploy(req)
+	if err != nil {
+		return err
+	}
+
+	return d.waitOnJobHelper(c, jobID, "creating", func(deployID string) error {
+		deploy, err := d.client.ReadDeploy(deployID)
+		if err != nil {
+			return err
+		}
+
+		return d.printer.PrintDeploys(deploy)
+	})
+}
+
+func (d *DeployCommand) delete(c *cli.Context) error {
+	return d.deleteHelper(c, "deploy", func(deployID string) (string, error) {
+		return d.client.DeleteDeploy(deployID)
+	})
+}
+
+func (d *DeployCommand) read(c *cli.Context) error {
+	args, err := extractArgs(c.Args(), "NAME")
+	if err != nil {
+		return err
+	}
+
+	deployIDs, err := d.resolver.Resolve("deploy", args["NAME"])
+	if err != nil {
+		return err
+	}
+
+	deploys := make([]*models.Deploy, len(deployIDs))
+	for i, deployID := range deployIDs {
+		deploy, err := d.client.ReadDeploy(deployID)
+		if err != nil {
+			return err
+		}
+
+		deploys[i] = deploy
+	}
+
+	return d.printer.PrintDeploys(deploys...)
+}
+
+func (d *DeployCommand) list(c *cli.Context) error {
+	deploySummaries, err := d.client.ListDeploys()
+	if err != nil {
+		return err
+	}
+
+	if !c.Bool("all") {
```

Comment:
I'm curious if this is going to cause unintended behavior given the provider is now only showing [active task definitions](https://github.com/quintilesims/layer0/blob/cli-refactor/api/provider/aws/deploy_list.go#L55) and [active task definition families](https://github.com/quintilesims/layer0/blob/cli-refactor/api/provider/aws/deploy_list.go#L38). Would a user expect to get a list of inactive task definitions if this flag is specified?

Reply:
:white_check_mark: